pytorch_lightning/core/lightning.py

```diff
@@ -112,8 +112,11 @@ def __init__(self, *args, **kwargs):
         self._current_hook_fx_name = None
         self._current_dataloader_idx = None
 
-    def optimizers(self):
-        opts = self.trainer.optimizers
+    def optimizers(self, use_pl_optimizer: bool = True):
+        if use_pl_optimizer:
+            opts = [*self.trainer.lightning_optimizers.values()]
```

Comment:
```suggestion
            opts = list(self.trainer.lightning_optimizers.values())
```
no need to expand and pack...

Reply:
done